cmd/oras/root/attach.go

```diff
@@ -164,7 +164,13 @@ func runAttach(ctx context.Context, opts attachOptions) error {
 
 	root, err := pushArtifact(dst, pack, copy)
 	if err != nil {
-		return err
+		if strings.Contains(err.Error(), "failed to delete dangling referrers index") {
```

Comment:
this looks fragile. Is there anything we can do to pushArtifact or the underlying code to return something else or ignore it at that level?

Reply:
Indeed it's fragile. Created an issue on SDK side https://github.com/oras-project/oras-go/issues/479

Reply:
Given that notation is also blocked on this, my vote would be to focus on returning a referrers manifest delete failure and handle that here instead of string checking.